File: android/daggerdemo/app/src/main/java/com/example/fcl/daggerdemo/Main2Activity.java
package com.example.fcl.daggerdemo;

import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.util.Log;
import android.view.View;
import android.widget.Toast;
import dagger.android.AndroidInjection;
import dagger.android.support.AndroidSupportInjection;
import dagger.android.support.DaggerAppCompatActivity;
import javax.inject.Inject;

public class Main2Activity extends DaggerAppCompatActivity {
    String TAG=Main2Activity.class.getSimpleName();
    @Inject
    Student student;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        AndroidInjection.inject(this);
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main2);
    }

    public void click(View view) {
        Log.e(TAG, student.getName());
        Toast.makeText(this, student.getName(), Toast.LENGTH_SHORT).show();
    }
}

File: android/LeakDemo/app/src/main/java/com/example/fcl/leakdemo/Main2Activity.kt
package com.example.fcl.leakdemo

import android.content.Intent
import android.support.v7.app.AppCompatActivity
import android.os.Bundle
import kotlinx.android.synthetic.main.activity_main2.jumpView

class Main2Activity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main2)
        jumpView.setOnClickListener {
            startActivity(Intent(this, MainActivity::class.java))
        }
    }
}
